src/Tribe/Integrations/Divi/Service_Provider.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Handles The Events Calendar integration with the Divi theme builder.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Integrations\Divi
+ */
+
+namespace Tribe\Events\Integrations\Divi;
+
+/**
+ * Class Service_Provider
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Integrations\Divi
+ */
+class Service_Provider extends \tad_DI52_ServiceProvider {
+
+	/**
+	 * Binds and sets up implementations.
+	 *
+	 * @since TBD
+	 */
+	public function register() {
+		$theme = wp_get_theme();
+
+		if ( 'divi' !== strtolower( $theme->name ) && 'divi' !== strtolower( $theme->parent_theme ) ) {
+			return;
+		}
+
+		$this->hooks();
+	}
+
+	/**
+	 * Hooks the filters and actions required for this integration to work.
+	 *
+	 * @since TBD
+	 */
+	public function hooks() {
+		add_filter(
+			'tribe_post_id',
+			[
+				$this,
+				'filter_tribe_post_id'
+			]
+		);
+	}
+
+	/**
+	 * Get the $event_id using get_queried_object_id() for Divi users who aren't using the Default Events Template.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $event_id The event ID.
+	 */
+	public function filter_tribe_post_id( $event_id ) {
+		// try the "normal" way first.
+		if ( empty( $event_id ) ) {
+			$event_id = get_the_ID();
+		}
+
+		// look for a post
+		if ( ! tribe_is_event( $event_id ) && isset( $_POST['et_post_id'] ) ) {
+			$event_id = $_POST['et_post_id'];
+		}
```

Comment:
`tribe_get_request()`?

Reply:
I'm not familiar with this method and I can't find it in TEC. Could you please provide a suggestion?

Reply:
He means use `tribe_get_request_var( 'et_post_id' )`

https://github.com/the-events-calendar/tribe-common/blob/master/src/functions/utils.php#L127

Reply:
@Camwyn Ooh I see, I've updated it.